typings/pinejs-client-core.d.ts

```diff
@@ -114,30 +117,53 @@ type BaseExpandFor<T> = { [k in keyof T]?: object } | StringKeyof<T>;
 
 export type Expand<T> = BaseExpandFor<T> | Array<BaseExpandFor<T>>;
 
-export interface PineOptions {
-	$select?: string[] | string | '*';
+export type ODataMethod = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+
+interface ODataOptionsBase {
+	$orderby?: OrderBy;
+	$top?: string;
+	$skip?: string;
+	$select?: string | string[] | '*';
+}
+
+export interface PineOptions extends ODataOptionsBase {
 	$filter?: object;
 	$expand?: object | string;
-	$orderby?: OrderBy;
-	$top?: number;
-	$skip?: number;
 }
 
-export interface PineOptionsFor<T> extends PineOptions {
+export interface PineOptionsFor<T> extends ODataOptionsBase {
 	$select?: Array<StringKeyof<T>> | StringKeyof<T> | '*';
 	$filter?: Filter<T>;
-	$expand?: Expand<T>;
+	$expand?: Expand<T> | string;
 }
 
-export interface PineParams {
-	resource: string;
+interface PineParamsBase {
+	apiPrefix?: string;
+	method?: ODataMethod;
+	resource?: string;
 	id?: number;
-	body?: object;
+	url?: string;
+	passthrough?: AnyObject;
+	passthroughByMethod?: {
+		GET: AnyObject;
+		POST: AnyObject;
```

Comment:
No PUT?

Reply:
I'm not sure, I just got that from pinejs-client
See: https://github.com/balena-io/pinejs-client-js/blob/7e582491765f9037a75754dfa2d0227e3365de33/core.ts#L1089

Reply:
@JackuB checked up on this with Page and it ends up that PUT should indeed be there and as a result [this PR was opened](https://github.com/balena-io/pinejs-client-js/pull/69).
Mirrored those changes in here.